index.js

```diff
@@ -121,22 +124,21 @@ Liftoff.prototype.buildEnvironment = function (opts) {
     }, this);
   }
 
-  // use rechoir to autoload any required modules
-  var autoloads;
-  if (configPath) {
-    autoloads = rechoir.prepare(this.extensions, configPath, cwd, true);
-    if (autoloads instanceof Error) {
-      autoloads = autoloads.failures;
-    }
-    if (Array.isArray(autoloads)) {
-      autoloads.forEach(function (attempt) {
-        if (attempt.error) {
-          this.emit('requireFail', attempt.moduleName, attempt.error);
-        } else {
-          this.emit('require', attempt.moduleName, attempt.module);
+  registerLoader(this.extensions, configPath, cwd, this);
+
+  var configFiles = {};
+  if (isPlainObject(this.configFiles)) {
+    var exts = this.extensions, notfound = { path: null }, eventEmitter = this;
+    configFiles = mapValues(this.configFiles, function(prop, name) {
+      var defaultObj = { name: name, cwd: cwd, extensions: exts };
+      return mapValues(prop, function(pathObj) {
+        var found = fined(pathObj, defaultObj) || notfound;
+        if (isPlainObject(found.extension)) {
```

Comment:
`found.extension` is either a string or a plain object which contains only an extension of a file which is found first.